### PR TITLE
test: introduce IT test for PromuSlice

### PR DESCRIPTION
**Pull request by baudoliver7**
We test availability of prometheus web service according to metrics configuration.

Ref: #980

### CODE REVIEW

**Comment by baudoliver7** [comment]
@artipie/maintainers ready to be reviewed :)

File: src/test/java/com/artipie/PromuSliceITCase.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            Yaml.createYamlMappingBuilder()
+                .add(
+                    "metrics",
+                    Yaml.createYamlMappingBuilder()
+                        .add("type", "prometheus")
+                        .build()
+                )
+        );
+        final HttpURLConnection con = (HttpURLConnection) new URL(formatUrl(port))
+            .openConnection();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/PromuSliceITCase.java:45]: Does not `HttpURLConnection` implement closable? If so, I'd suggest to use `try-finally` block here.

File: src/test/java/com/artipie/PromuSliceITCase.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Test
+    void blocksAccessWhenPrometheusConfigIsUnavailable(
+        @TempDir final Path root) throws Exception {
+        final int port = runServer(
+            root,
+            Yaml.createYamlMappingBuilder()
+        );
+        final HttpURLConnection con = (HttpURLConnection) new URL(formatUrl(port))
+            .openConnection();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/PromuSliceITCase.java:65]: same here

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@baudoliver7 one suggestion, please, take a look. Also, all the ITs in artipie/artipie are implemented with the artipie docker container usage, maybe here we should do the same. @g4s8 what do you think?

File: src/test/java/com/artipie/PromuSliceITCase.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/PromuSliceITCase.java:45]: I'm afraid not.

File: src/test/java/com/artipie/PromuSliceITCase.java (hunk 2 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/PromuSliceITCase.java:65]: ditto

**Comment by baudoliver7** [comment]
@olenagerasimova I understand the practice here in Artipie. However, I think IT testing is more about bringing multiple subsystems (classes or modules) into play to test a behavior.
In my case, I run an Artipie server before testing availability of `Prometheus` web service according to metrics settings.

**Review by olenagerasimova — APPROVED** [review]
@baudoliver7 ok

**Comment by baudoliver7** [comment]
@olenagerasimova Thanks !

**Comment by baudoliver7** [comment]
@g4s8 please, check/merge

**Comment by baudoliver7** [comment]
@g4s8 I think you forgot to check/merge this.

**Comment by baudoliver7** [comment]
@g4s8 It has been approved. Please, check/merge

File: src/test/java/com/artipie/PromuSliceITCase.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                            .add("path", repos.toString())
+                            .build()
+                    ).build()
+                )
+                .build().toString().getBytes()
+        );
+        final JettyClientSlices http = new JettyClientSlices(new HttpClientSettings());
+        http.start();
+        final VertxMain server = new VertxMain(http, cfg, vertx, 0);
+        return server.start();
```
Comment (g4s8) [inline src/test/java/com/artipie/PromuSliceITCase.java:107]: Each resource should be started in `setup` and stopped in `tearDown` test method - this way we can control the lifecycle. Right now, this test produces a few different servers without stopping it.

**Review by g4s8 — CHANGES_REQUESTED** [review]
@baudoliver7 pleas check my comments

File: src/test/java/com/artipie/PromuSliceITCase.java (hunk 3 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/PromuSliceITCase.java:107]: Right !

Comment (baudoliver7) [reply src/test/java/com/artipie/PromuSliceITCase.java:107]: Besides, how about using `testcontainers` as @olenagerasimova mentioned early ?

**Comment by baudoliver7** [comment]
@olenagerasimova @g4s8 I think it's good now :)

**Comment by baudoliver7** [comment]
@g4s8 I think latest changes make this a good candidate to be merged :)

**Review by g4s8 — DISMISSED** [review]
@baudoliver7 thanks

**Comment by baudoliver7** [comment]
@g4s8 please, try to merge again

**Comment by g4s8** [comment]
@baudoliver7 now this PR contains a lot of unrelated changes

**Comment by baudoliver7** [comment]
@g4s8 branch cleaned now :)
Besides, I suggest you to make `pdd-lint` required to avoid that a slipping error prevent `pdd` to generate new puzzles.

**Review by g4s8 — APPROVED** [review]
@baudoliver7 thanks

**Pull request merged**